release/models/platform/openconfig-platform-fabric.yang

```diff
@@ -0,0 +1,88 @@
+module openconfig-platform-fabric {
+
+  yang-version "1";
+
+  // namespace
+  namespace "http://openconfig.net/yang/platform/fabric";
+
+  prefix "oc-fabric";
+
+  import openconfig-platform { prefix oc-platform; }
+  import openconfig-platform-types { prefix oc-platform-types; }
+  import openconfig-extensions { prefix oc-ext; }
+
+
+  // meta
+  organization "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This module defines data related to FABRIC components in
+    the openconfig-platform model";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2022-04-20" {
+    description
+      "Initial revision";
+    reference "0.1.0";
+  }
+
+  // OpenConfig specific extensions for module metadata.
+  oc-ext:regexp-posix;
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  // extension statements
+
+  // feature statements
+
+  // identity statements
+
+  // typedef statements
+
+  // grouping statements
+
+  grouping fabric-config {
+    description
+      "Configuration data for fabric components";
+
+    leaf power-admin-state {
```

Comment:
Should we make this a common definition somewhere? It could probably be a `typedef` in `openconfig-platform-types`.

Reply:
Thanks for the catch. I'll remove this from both files - openconfig-platform-fabric.yang and openconfig-platform-controller-card.yang

Reply:
I'm not sure if I understand this correctly -- did you mean `grouping` rather than `typedef`? I  can define the `grouping` in `openconfig-platform-types` so that we can code re-use it in all three modules: linecard, fabric and controller-card

Reply:
If this is going to be added shouldn't there be other components that it should cover to be consistent such as for transceiver and power-supply that both have an "enabled" for power on/off and potentially others.  Would it make sense if we want to add this to add it under the /components/component/state path?

Reply:
@cfyo I think this is not a bad idea, but is perhaps some minor scope creep for this PR.    The other defined OC component types I think this might make sense for are fan and power supply?   For power supply, it might be more appropriate to support an admin-config leaf for 'enabled/disabled'.   Transceiver also has enabled/disabled which probably controls its electrical power?   May I suggest we address that in a separate PR so we can keep this one moving forward?